### PR TITLE
httptransport: use ctxhttp to propagate context

### LINKED ISSUE #154: HTTP client transport should use ctxhttp?
Since https://github.com/golang/go/issues/11904, the context package has a nice wrapper around http.Client that is context aware. Would there be any interest in a PR that uses that package internally to wrap the http.Client so `cancel()`s will propagate to the request?

### PR DESCRIPTION
**Pull request by simonjefford**
Fixes #154

### CODE REVIEW

**Comment by peterbourgon** [comment]
That easy, huh? :)

File: transport/http/client.go [hunk 1]
```diff
@@ -5,9 +5,9 @@ import (
 	"net/http"
 	"net/url"
 
-	"golang.org/x/net/context"
-
 	"github.com/go-kit/kit/endpoint"
+	"golang.org/x/net/context"
+	"golang.org/x/net/context/ctxhttp"
```
Comment (peterbourgon) [inline transport/http/client.go]: Please keep these imports where they were before, above the go-kit block.

**Comment by peterbourgon** [comment]
That minor nit aside, :+1:

**Comment by simonjefford** [comment]
Apologies - force of habit - I tend to delete blank lines in the "3rd party" import block then let `goimports` sort them.

**Pull request merged**